client/components/AllProducts.js

```diff
@@ -1,25 +1,40 @@
 import React, {Component} from 'react'
 import {connect} from 'react-redux'
+import {fetchProducts} from '../store/product'
 
 class AllProducts extends Component {
   componentDidMount() {
-    // TODO: Uncomment line below when we have a thunk imported from the store named fetchProducts.
-    // this.props.fetchProducts();
+    this.props.fetchProducts()
   }
 
   render() {
     const {products} = this.props
 
     return (
       <div>
-        {products ? <div>{products}</div> : <div>No products in database.</div>}
+        {products.length ? (
+          <div>
+            <h2>Products Available</h2>
+            {products.map(product => {
+              return (
+                <div key={product.id}>
+                  <h3>{product.title}</h3>{' '}
+                  <p>Price: ${product.priceCents / 100}</p>
+                  <p>Description: {product.description}</p>
+                </div>
+              )
+            })}
+          </div>
+        ) : (
+          <div>No products in database.</div>
+        )}
       </div>
     )
   }
 }
 
 const mapStateToProps = state => ({
-  products: state.products
+  products: state.productsReducer
```

Comment:
Why did this change from `products` (a noun) to `productsReducer` (which sounds like a reducer, aka a function)?

Reply:
It's not a big deal but it seems like an odd name choice, the state in your Redux would normally be a bunch of nouns. Like in your combine reducers you'd do:

```js
const fullReducer = combineReducers({
    someNoun: someNounReducer,
    otherNoun: otherNounReducer
})
```

This can be changed in a separate PR – make lots of frequent small PRs rather than big multi-purpose PRs.